addons/sale/models/sale_order.py

```diff
@@ -404,21 +404,16 @@ def _compute_team_id(self):
                     user_id=user_id, domain=[('company_id', 'in', [company_id, False])])
             order.team_id = cached_teams[key]
 
-    @api.depends('order_line.price_total')
+    @api.depends('order_line.price_subtotal', 'order_line.price_tax', 'order_line.price_total')
     def _amount_all(self):
         """
         Compute the total amounts of the SO.
         """
         for order in self:
-            amount_untaxed = amount_tax = 0.0
-            for line in order.order_line:
-                amount_untaxed += line.price_subtotal
-                amount_tax += line.price_tax
-            order.update({
-                'amount_untaxed': amount_untaxed,
-                'amount_tax': amount_tax,
-                'amount_total': amount_untaxed + amount_tax,
-            })
+            order_lines = order.order_line.filtered(lambda x: not x.display_type)
+            order.amount_untaxed = sum(order_lines.mapped('price_subtotal'))
+            order.amount_total = sum(order_lines.mapped('price_total'))
+            order.amount_tax = order.amount_total - order.amount_untaxed
```

Comment:
Mind explaining/adding a comment why the `amount_tax` is computed like that and not `sum(order_lines.mapped('price_tax'))` (or like before amount_tax + untaxed) ?

I find the change of behavior confusing (and even more when the logic is different between the SOL and SO amount computations) :thinking:

Reply:
Both should give the same result :) I change for `price_tax` if you like

Reply:
According to https://github.com/odoo/odoo/pull/87387, it seems that it might have an impact :D (feel free to comment if you want).

I was curious about the reason behind the change, if there was a specific reason or not.
If there isn't, I'd say it's better to follow the line computation as much as possible, and use:
```
order.amount_tax = sum(order_lines.mapped('price_tax'))
order.amount_total = order.amount_untaxed + order.amount_tax
```
That way, even if there are rounding changes, the total will be the correct sum of the untaxed amount + taxes.

Reply:
@Feyensv This commit is fixing the unconsistencies between SO / invoices when using the "round globally" method. This will not be an issue anymore after the merge :)

Reply:
Perfect then :+1: